### PR TITLE
Curve review

### PR DESCRIPTION
**Pull request by fabiohild**

### CODE REVIEW

**Review by michwill — COMMENTED** [review]
Two comments.

a) Can we keep sqrt inside the smart contract? This will save gas, and really makes it easier to check.

b) Can we use the second method (which is much more stable), e.g. where you divide virtual price by sqrt(coinPrice)? (if I remember correctly)

**Comment by fabiohild** [comment]
Hey @michwill , thanks for the review. I updated the virtual price 2 formula and brought the sqrt function into the contract.

I actually tried that at the first time, but ran into the EIP170 contract size limit. I removed the donate_admin_fees function to accommodate, let me know if this works.

**Comment by amadeobrands** [comment]
Looks good to me.
A nice kick start of what is to come with the Curve V2 Pools.

File: contracts/pools/rai/StableSwapRAI.vy [hunk 1]
```diff
@@ -98,14 +98,14 @@ event StopRampA:
     t: uint256
 
 
-N_COINS: constant(int128) = 2
-MAX_COIN: constant(int128) = N_COINS - 1
-REDEMPTION_COIN: constant(int128) = 0  # Index of asset with moving target redemption price
-REDMPTION_PRICE_SCALE: constant(int128) = 10 ** 9
+N_COINS: constant(uint256) = 2
```
Comment (halito22) [inline contracts/pools/rai/StableSwapRAI.vy:100]: thank you

File: contracts/pools/rai/StableSwapRAI.vy [hunk 2]
```diff
@@ -98,14 +98,14 @@ event StopRampA:
...
+MAX_COIN: constant(uint256) = N_COINS - 1
+REDEMPTION_COIN: constant(uint256) = 0  # Index of asset with moving target redemption price
+REDMPTION_PRICE_SCALE: constant(uint256) = 10 ** 9
```
Comment (halito22) [inline contracts/pools/rai/StableSwapRAI.vy:103]: thank you

File: contracts/pools/rai/StableSwapRAI.vy [hunk 3]
```diff
@@ -98,14 +98,14 @@ event StopRampA:
...
 
 FEE_DENOMINATOR: constant(uint256) = 10 ** 10
 PRECISION: constant(uint256) = 10 ** 18  # The precision to convert to
-BASE_N_COINS: constant(int128) = 3
+BASE_N_COINS: constant(uint256) = 3
```
Comment (halito22) [inline contracts/pools/rai/StableSwapRAI.vy:107]: thank you

**Review by halito22 — APPROVED** [review]

**Pull request merged**